src/main/java/net/snowflake/client/core/JsonSqlOutput.java

```diff
@@ -0,0 +1,390 @@
+        }));
+  }
+
+  @Override
+  public void writeDouble(double value) throws SQLException {
+    withNextValue(
+        ((json, fieldName, maybeColumn) -> {
+          json.put(fieldName, value);
+          schema
+              .getFields()
+              .add(buildSchemaWithScaleAndPrecision(fieldName, "real", 0, 0, maybeColumn));
+        }));
+  }
+
+  @Override
+  public void writeBigDecimal(BigDecimal value) throws SQLException {
+    withNextValue(
+        ((json, fieldName, maybeColumn) -> {
+          json.put(fieldName, value);
+          schema
+              .getFields()
+              .add(buildSchemaWithScaleAndPrecision(fieldName, "real", 0, 0, maybeColumn));
+        }));
+  }
+
+  @Override
+  public void writeBytes(byte[] value) throws SQLException {
+    withNextValue(
+        ((json, fieldName, maybeColumn) -> {
+          json.put(fieldName, new SFBinary(value).toHex());
+          schema
+              .getFields()
+              .add(FieldSchemaCreator.buildSchemaForBytesType(fieldName, maybeColumn));
+        }));
+  }
+
+  @Override
+  public void writeDate(Date value) throws SQLException {
+    withNextValue(
+        ((json, fieldName, maybeColumn) -> {
+          json.put(
+              fieldName,
+              ResultUtil.getDateAsString(value, getDateTimeFormat("DATE_OUTPUT_FORMAT")));
+          schema
+              .getFields()
+              .add(FieldSchemaCreator.buildSchemaTypeAndNameOnly(fieldName, "date", maybeColumn));
+        }));
+  }
+
+  @Override
+  public void writeTime(Time x) throws SQLException {
+    withNextValue(
+        ((json, fieldName, maybeColumn) -> {
+          final long MS_IN_DAY = 86400 * 1000;
+          long msSinceEpoch = x.getTime();
+          long msSinceMidnight = (msSinceEpoch % MS_IN_DAY + MS_IN_DAY) % MS_IN_DAY;
+          long nanosSinceMidnight = msSinceMidnight * 1000 * 1000;
+          String result =
+              ResultUtil.getSFTimeAsString(
+                  SFTime.fromNanoseconds(nanosSinceMidnight),
+                  9,
+                  getDateTimeFormat("TIME_OUTPUT_FORMAT"));
+
+          json.put(fieldName, result);
+          schema
+              .getFields()
+              .add(buildSchemaWithScaleAndPrecision(fieldName, "time", 9, 0, maybeColumn));
+        }));
+  }
+
+  @Override
+  public void writeTimestamp(Timestamp value) throws SQLException {
+    withNextValue(
+        ((json, fieldName, maybeColumn) -> {
+          SnowflakeType snowflakeType =
+              SnowflakeType.fromString(
+                  maybeColumn
+                      .map(cl -> cl.type())
+                      .filter(str -> !str.isEmpty())
+                      .orElse("TIMESTAMP"));
+          int columnType = SnowflakeUtil.EXTRA_TYPES_TIMESTAMP_TZ;
+          TimeZone timeZone = TimeZone.getDefault();
+          if (snowflakeType == SnowflakeType.TIMESTAMP_NTZ
+              || snowflakeType == SnowflakeType.TIMESTAMP) {
```

Comment:
This one depends on [TIMESTAMP_TYPE_MAPPING](https://docs.snowflake.com/en/sql-reference/parameters.html#label-timestamp-type-mapping) or not?

Reply:
In SnowflakeType.java we have 4 types for tyimestamp';  TIMESTAMP,
  TIMESTAMP_LTZ,
  TIMESTAMP_NTZ,
  TIMESTAMP_TZ,

Reply:
Yes, but we can change meaning of `TIMESTAMP` from being `TIMESTAMP_NTZ` to other variant using this parameter. But we should investigate in jdbc globally if it is handled correctly...